### PR TITLE
Embed CFamily analyzer 6.23.0.34138

### LINKED ISSUE #2521: Embed CFamily analyzer 6.23.0.34138
#### Description

For CMake support we need to embed CFamily analyzer 6.22.0.33947 or higher. The latest version is currently 6.23.0.34138.

### PR DESCRIPTION
**Pull request by rita-sonarsource**
Fixes #2521

### CODE REVIEW

File: src/Integration.Vsix.UnitTests/CFamily/MsvcDriverTest.cs [hunk 1]
```diff
@@ -760,5 +760,80 @@ public void AbsolutePathConversion()
...
+        [DataRow("19.11.25507", "/permissive- /Zc:twoPhase-", false)]
+        [DataRow("19.28.29333", "", false)] // No-permissive is default for latest
+        [DataRow("19.28.29333", "/permissive-", true)]
+        [DataRow("19.28.29333", "/std:c++latest", true)]
+        [DataRow("19.28.29333", "/permissive", false)]
+        [DataRow("19.28.29333", "/Zc:twoPhase-", false)]
+        [DataRow("19.28.29333", "/permissive- /Zc:twoPhase-", false)]
+        [DataRow("19.28.29333", "/std:c++latest /Zc:twoPhase-", false)]
+        [DataRow("19.28.29333", "/std:c++latest /permissive /Zc:twoPhase-", false)]
+        public void NoDelayedTemplateParsing(string compilerVersion, string cmd, bool expected)
```
Comment (rita-sonarsource) [inline src/Integration.Vsix.UnitTests/CFamily/MsvcDriverTest.cs:780]: Ported from https://github.com/SonarSource/sonar-cpp/blob/3dc9a31ba6a8d66e2e113c2d64290ef0e23cf9ae/sonar-cfamily-plugin/src/test/java/com/sonar/cpp/analyzer/MsvcDriverTest.java#L442

File: src/Integration.Vsix.UnitTests/CFamily/MsvcDriverTest.cs [hunk 2]
```diff
@@ -760,5 +760,80 @@ public void AbsolutePathConversion()
...
+        [TestMethod]
+        [DataRow("/std:c++17", true)] // enabled by default
+        [DataRow("/std:c++latest", true)] // enabled by default
+        [DataRow("/std:c++17 /Zc:alignedNew", true)]
+        [DataRow("/std:c++latest /Zc:alignedNew", true)]
+        [DataRow("/std:c++17 /Zc:alignedNew-", false)]
+        [DataRow("/std:c++latest /Zc:alignedNew-", false)]
+        [DataRow("/std:c++14 /Zc:alignedNew", false)] // not supported in old languages
+        [DataRow("/Zc:alignedNew", false)] // not supported in old languages
+        public void AlignedNew(string cmd, bool expected)
```
Comment (rita-sonarsource) [inline src/Integration.Vsix.UnitTests/CFamily/MsvcDriverTest.cs:816]: Ported from https://github.com/SonarSource/sonar-cpp/blob/3dc9a31ba6a8d66e2e113c2d64290ef0e23cf9ae/sonar-cfamily-plugin/src/test/java/com/sonar/cpp/analyzer/MsvcDriverTest.java#L325
and
https://github.com/SonarSource/sonar-cpp/blob/3dc9a31ba6a8d66e2e113c2d64290ef0e23cf9ae/sonar-cfamily-plugin/src/test/java/com/sonar/cpp/analyzer/MsvcDriverTest.java#L351

File: src/Integration.Vsix/CFamily/PortedFromJava/MsvcDriver.cs [hunk 3]
```diff
@@ -430,6 +454,24 @@ public static Request ToRequest(CFamilyHelper.Capture[] captures)
                     {
                         request.Flags |= Request.OperatorNames;
                     }
+                    if ((request.Flags & Request.CPlusPlus17) != 0 && !alignedNewDisabled)
+                    {
+                        request.Flags |= Request.AlignedNew;
+                    }
+
+                    var cppLatestImpliesUnpermissive = (major > 19) || (major == 19 && minor >= 28);
```
Comment (rita-sonarsource) [inline src/Integration.Vsix/CFamily/PortedFromJava/MsvcDriver.cs:462]: Ported from https://github.com/SonarSource/sonar-cpp/blob/3dc9a31ba6a8d66e2e113c2d64290ef0e23cf9ae/sonar-cfamily-plugin/src/main/java/com/sonar/cpp/analyzer/MsvcDriver.java#L335

The original code uses a "version" class; I extracted the methods from there, keeping the original names

**Review by abbas-sabra-sonarsource — APPROVED** [review]
LGTM. Thanks for handling it!

**Review by duncanp-sonar — APPROVED** [review]

**Pull request merged**